test/mocks/server/server_factory_context.h

```diff
@@ -95,6 +96,7 @@ class MockServerFactoryContext : public virtual ServerFactoryContext {
   testing::NiceMock<Upstream::MockClusterManager> cluster_manager_;
   testing::NiceMock<Event::MockDispatcher> dispatcher_;
   testing::NiceMock<MockDrainManager> drain_manager_;
+  TransportSocketFactoryContext* transport_socket_factory_context_;
```

Comment:
I didn't use `testing::NiceMock<MockTransportSocketFactoryContext>`, because that caused circular dependency error.